k8s/README.md

```diff
@@ -0,0 +1,74 @@
+# Kubernetes
+
+This directory covers everything related to Kubernetes.
+
+## LitterRally Deployment
+
+The application in both dev and prod environments are deployed through ArgoCD.
+
+Pre-requisites:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- A kubernetes cluster (minikube, GKE, etc)
+
+### Deployment Steps
+
+1. Deploy ArgoCD
+
+   ArgoCD will be the controller who will watch over our application environment states, and ensures that the live kubernetes resources are truly matched to the GitHub repository kubernetes manifests (GitOps).
+
+   ```
+   $ kubectl create namespace argocd
+   $ kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+
+   # If on GKE, give yourself permission to deploy cluster roles
+   $ gcloud auth login
+   $ kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user="$(gcloud config get-value account)"
+   ```
+
+2. Deploy Dev and Prod Environments
+
+   Now that ArgoCD is deployed, we need to tell ArgoCD how exactly to control/watch these kubernetes manifests.
+
+   ```
+   $ cd k8s/argocd-apps
+   $ kubectl apply -f .
+   ```
+
+3. Login to ArgoCD
+
+   - On minikube/local cluster:
+
+     1. Get the password by getting the argocd-repo-server pod name
+
+        `$ kubectl get pods -n argocd -l app.kubernetes.io/name=argocd-server -o name | cut -d'/' -f 2`
+
+     2. Port-forward the argocd service
+
+        `$ kubectl port-forward svc/argocd-server -n argocd 8080:443`
+
+     3. Go to `localhost:8080` and set
+        ```
+        user: admin
+        password: the pod name you get from step 1
+        ```
+
+   - On GKE:
+
+     TODO: Create safe login workflow (for now, contact Ritchelle please)
+
+4. Sync Applications
+
+   Once logged in to ArgoCD, click `SYNC` on each application environment to sync the ArgoCD applications, which will finally deploy the LitterRally applications.
+
+5. Access the Applications
+
+   We currently don't have an ingress installed, so we have to port-forward the frontend and backend.
+
+   ```
+   $ export LITTERRALLY_POD=$(kubectl get pods -n litterrally-[dev|prod] | awk '{print $1}' | sed -n '2 p')
+   $ kubectl -n litterrally-[dev|prod] port-forward $LITTERRALLY_POD 9000:8080
+   $ kubectl -n litterrally-[dev|prod] port-forward $LITTERRALLY_POD 3333:3333
```

Comment:
First I had some issues getting this to work since the actual port for the front-end for our current pod is 80, not 8080.

Second, I managed to get it to work with something simpler:
```suggestion
   $ export LITTERRALLY_POD=$(kubectl get pods -n litterrally-[dev|prod] | awk '{print $1}' | sed -n '2 p')
   $ kubectl -n litterrally-[dev|prod] port-forward $LITTERRALLY_POD 9000:80 3333
```

Reply:
Whoops yeah dangers of copy paste, I initially had it hitting the service and decided to change to the pod, but forgot to update the port 👍

Reply:
Also didn't know the second way was a thing to begin with, TIL!